### ISSUE TITLE
[BUG/ISSUE] Bugfix for Get_IJ routine in gc_grid_mod affecting gosat operator

### ISSUE THREAD

**Issue by sabourbaray** [geoschem/geos-chem#1607]
The routine Get_IJ in gc_grid_mod is used by the gosat operator (as well as TCCON, planeflight diagnostics etc.) to get the grid box corresponding to the observations. I think there are two lines that should be corrected, this bug appears when working with an irregularly shaped grid (such as with the IMI system) or when observations are at the boundaries of the nested domain.

1. Line 977-978 converts LON/LAT to TLON/TLAT. In this conversion "+1.5" is added. As I understand, the point here is that the first grid box should be 1, but I don't know why an additional half-grid box adjustment is applied. Changing +1.5 to +1.0 corrected getting a "Beyond the nested window" error for me.

L977-978:
```
     TLON = INT( ( LON + 180e+0_fp ) / State_Grid%DX + 1.5e+0_fp )
     TLAT = INT( ( LAT +  90e+0_fp ) / State_Grid%DY + 1.5e+0_fp )
```
Changed to:
```
     TLON = INT( ( LON + 180e+0_fp ) / State_Grid%DX + 1.0e+0_fp )
     TLAT = INT( ( LAT +  90e+0_fp ) / State_Grid%DY + 1.0e+0_fp )
```

2. Line 983 compares TLAT to State_Grid%NX which I think should be State_Grid%NY:
```
     IF ( TLAT < 1 .or. TLAT > State_Grid%NX ) THEN
```
Changed to:
```
     IF ( TLAT < 1 .or. TLAT > State_Grid%NY ) THEN
```

Please let me know if this is correct. Thanks!

**Comment by yantosca** [comment]
Thanks @sabourbaray. We will look into this.

**Comment by yantosca** [comment]
Also tagging @msulprizio who is more familiar w/ the observational operators code.

**Comment by yantosca** [comment]
Also I think the 0.5 is added to round up to the nearest integer.  For example `INT( X )` is the same as `FLOOR( X )` but `INT( X + 0.5 )` is the same as `NINT( X )`.

**Comment by yantosca** [comment]
This is now included in 14.1.1.  We can close this issue out.
